sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -598,6 +598,16 @@ object SQLConf {
       .bytesConf(ByteUnit.BYTE)
       .createOptional
 
+  val ADAPTIVE_SHUFFLE_HASH_JOIN_LOCAL_MAP_THRESHOLD =
+    buildConf("spark.sql.adaptive.shuffledHashJoinLocalMapThreshold")
+      .doc("Configures the maximum size in bytes for per partition that can be allowed to build " +
+        "local hash map. If all the partition size not larger than this threshold, join " +
+        "selection may use shuffled hash join instead of sort merge join regardless of the " +
```

Comment:
I feel `may` is quite vague here. Can't we just say `prefer to use shuffled hash join instead of sort merge join regardless of ...`